tls/test_tls_cert.py

```diff
@@ -608,17 +607,8 @@ def handshake(sni):
             generate_certificate(san=next(san_iter))
             self.get_tempesta().reload()
 
-            try:
-                handshake(next(sni_iter))
-            except tls.TLSProtocolError:
-                raise Exception(f"SNI should match to the current certificate [i={i}]")
-
-            with self.assertRaises(
-                tls.TLSProtocolError, msg=f"SNI should not match to the current certificate [i={i}]"
-            ):
-                handshake(next(sni_iter))
-
-            next(sni_iter)  # additional shift to alternate the order
+            self.assertTrue(handshake(next(sni_iter)), "First handshake should pass")
+            self.assertFalse(handshake(next(sni_iter)), "Second handshake should fail")
 
```

Comment:
`next(sni_iter)  # additional shift to alternate the order`
line is missed at the end

Reply:
Added